infra/gcp/terraform/k8s-infra-prow-build/iam.tf

```diff
@@ -35,6 +35,6 @@ resource "google_iam_workload_identity_pool_provider" "eks_cluster" {
   oidc {
     # From EKS cluster created in https://github.com/kubernetes/k8s.io/tree/main/infra/aws/terraform/prow-build-cluster
     issuer_uri        = "https://oidc.eks.us-east-2.amazonaws.com/id/F8B73554FE6FBAF9B19569183FB39762"
-    allowed_audiences = ["https://sts.googleapis.com"]
+    allowed_audiences = ["sts.googleapis.com"]
```

Comment:
We should be explicit about protocol used here.

Reply:
Protocol isn't required. It is an arbitrary string that both systems have configured.

Reply:
The pod is creating accounts with aud `sts.googleapis.com`